src/pip/_internal/commands/completion.py

```diff
@@ -34,10 +34,8 @@
     """,
     'fish': """
         function __fish_complete_pip
-            set -lx COMP_WORDS (commandline -o) ""
-            set -lx COMP_CWORD ( \\
-                math (contains -i -- (commandline -t) $COMP_WORDS)-1 \\
-            )
+            set -lx COMP_WORDS (commandline -opc) (commandline -ct)
+            set -lx COMP_CWORD (math (count $COMP_WORDS) - 1)
```

Comment:
Hah, I was drafting a similar PR last night. Can I suggest changing to the long options? It makes it much clearer for people not familiar with fish.

Reply:
Yeah, probably better this way, although a bit verbose.